src/Mvc/Mvc.RazorPages/test/Infrastructure/PageActionEndpointDataSourceTest.cs

```diff
@@ -150,15 +150,14 @@ public void GroupedEndpoints_AppliesConventions_RouteSpecificMetadata()
             {
                 Assert.Equal("/group1/test", e.RoutePattern.RawText);
                 Assert.Same(actions[0], e.Metadata.GetMetadata<ActionDescriptor>());
-                // Assert.Equal(new[] { "group", "Hi there", "A" }, e.Metadata.GetOrderedMetadata<string>());
+                Assert.Equal(new[] { "A", "group", "Hi there" }, e.Metadata.GetOrderedMetadata<string>());
```

Comment:
This is wrong no? The group should be overridden by the local metadata entry?

Reply:
I had this inquiry as well when I originally wrote this but I eventually deduced that its because I'm registering the action-specific metadata differently in each scenario.

In the controller scenario, the local metadata is registered using a convention on the controller action builder. The route-specific conventions run after the global ones when the action endpoint is created.

In the Razor page scenario, the local metadata is provided in `ActionDescriptor.EndpointMetadata` which gets registered before the metadata that is added before conventions.

https://github.com/dotnet/aspnetcore/blob/c2cba55ca149ab3bd3fadcf97b9d16a74f561111/src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs#L315-L322

Now, to be honest, it seems weird to me that the code above would treat action-specific metadata as having a _lower_  precedence than other conventions, but it seems like this behavior has been around for a while.

For consistencies sake, I can update both tests to use the `Page`/`Controller` convention builders to register metadata.

Reply:
Actually, based on my understanding of the code, this makes sense for asserting the behavior that group metadata has lower precedence than local metadata. The `AddRoute` method used in the controller tests doesn't necessarily map for Razor pages since it is used to set fallback routes for controllers.

Reply:
As long as there's a test that verifies that adding metadata via say an attribute on the controller action isn't overridden by what a group adding that same attribute, then we're golden.

Reply:
This makes sense to me. Do we have the test for attributes yet? It might make sense to add a call to `MapGroup` inside `StartupForEndpointFilters` so we can test it end-to-end. This way we can test both adding filters via a group and adding other metadata, and the relative ordering of all of this.

Reply:
Also, I want to make sure any metadata applied to the outer group does have lower precedence than something added via something like `outerGroup.WithMetadatata("Before A").AddRoute("1", "/1/{controller}/{action}/{id?}", null, null, null).Add(b => b.Metadata.Add("A"));`

Reply:
Added a test here, this took me a while to understand but it seems that route specific entries override attributes on the endpoint.